src/usr/share/gpgmailer/gpgkeyverifier.py

```diff
@@ -163,33 +151,33 @@ def _calculate_recipient_info(self, loop_current_time):
 
             else:
                 # Always encrypt with the sender key. TODO: Eventually make this an option.
-                valid_key_fingerprints.append(self.email_dicts[self.sender_email]['fingerprint'])
-
-                if self.email_dicts[self.sender_email]['is_recipient']:
-                    valid_recipient_emails.append(self.sender_email)
+                valid_key_fingerprints.append(self.sender_key['fingerprint'])
+                for email in self.sender_key['email_list']:
+                    valid_recipient_emails.append(email)
 
             # The sender's message always shows up on top regardless of whether the key has expired
             #   or will expire soon.
             expired_messages.append(sender_expiration_data['warning_message'])
 
         else:
             # Always encrypt with the sender key. TODO: Eventually make this an option.
-            valid_key_fingerprints.append(self.email_dicts[self.sender_email]['fingerprint'])
-
-            if self.email_dicts[self.sender_email]['is_recipient']:
-                valid_recipient_emails.append(self.sender_email)
+            valid_key_fingerprints.append(self.sender_key['fingerprint'])
+            for email in self.sender_key['email_list']:
+                valid_recipient_emails.append(email)
 
+        # Calculate all of the expiration information for each recipient (i.e. non-sender keys).
         self.logger.trace('Checking recipient keys.')
-        for recipient_email in self.all_recipient_emails:
+        #
+        for fingerprint in self.gpgkeyring.fingerprint_to_key_dict.keys():
             # Reuse the sender expiration data if the sender is also a recipient.
             # TODO: Eventually, optimize all key checking, not just sender key. Maybe have a list
             #   of checked fingerprints.
```

Comment:
I don't think this TODO is relevant anymore.

Reply:
k